tests/lib/prepare-utils.sh

```diff
@@ -114,9 +116,20 @@ start_snapd_core_vm() {
     nested_wait_for_snap_command
 }
 
+get_arch() {
+    if os.query is-pc-amd64; then
+        printf amd64
+    elif os.query is-arm64; then
+        printf arm64
+    else
+        printf "ERROR: unsupported archtecture\n"
+        exit 1
+    fi
+}
+
 get_core_snap_name() {
     printf -v date '%(%Y%m%d)T' -1
-    echo "core24_${date}_amd64.snap"
+    echo "core24_${date}_$(get_arch).snap"
```

Comment:
I don't see where you do this for core22, I just want to make sure it is considered.